### PR TITLE
First pass at smooth scrolling

### PR DESCRIPTION
**Pull request by MattPutnam**
Needs some improvement, but works for now.

### CODE REVIEW

**Comment by rushtong** [comment]
This looks very cool! Would it be worth it to add an optional argument to scroll-to for the scroll time (currently defaults to 100)?

**Comment by MattPutnam** [comment]
Good idea.  I'll add that.

File: src/cljs/org/broadinstitute/firecloud_ui/common.cljs [hunk 1]
```diff
@@ -27,16 +27,46 @@
...
+      (.scrollTo js/window end-x end-y)
+      (let [point (smooth-step start-time end-time (js/Date.now))]
+        (.scrollTo js/window
+          (+ start-x (* point (- end-x start-x)))
+          (+ start-y (* point (- end-y start-y))))
+        (js/setTimeout #(animate start-time end-time start-x start-y end-x end-y) 10)))))
+
+(defn scroll-to
+  ([x y] (.scrollTo js/window x y))
+  ([x y duration]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common.cljs]: I suggest:
```Clojure
(assert (<= duration 400) "Duration too long (> 400ms)")
```
to ensure that the caller cannot accidentally create a bad experience for the user.

**Comment by dmohs** [comment]
:+1: This is, of course, fully awesome. Merge whenever.

**Pull request merged**